### PR TITLE
[#159254614] Check the background activity and if needed force to user to inser th…

### PR DESCRIPTION
**Pull request by fpersico**
…e PIN

NOTE: When the video was recorded the timeout was 15 sec.

background

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
## Affected stories

  * 🌟 [#159254614](https://www.pivotaltracker.com/story/show/159254614): Come APP, dopo 30 secondi da quando vengo messa in background, devo chiedere all'utente il passcode.

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by cloudify** [comment]
from the video it seems the messages screen appears for a split second before the unlock screen appears...

**Comment by fpersico** [comment]
Video update with the new version.

File: ts/navigation/AppNavigator.ts [hunk 1]
```diff
@@ -8,30 +9,37 @@ import PinNavigator from "./PinNavigator";
...
- * SwitchNavigator is very useful here because it automatically
- * resets the state on navigation
+ * The main stack of screens of the Application.
  */
-const navigator = createSwitchNavigator({
-  [ROUTES.INGRESS]: {
-    // This screen check if the user is authenticated than perform a redirect to
-    // MainNavigator (if authenticated) or AuthenticationNavigator (otherwise)
-    screen: IngressScreen
+const navigator = createStackNavigator(
```
Comment (cloudify) [inline ts/navigation/AppNavigator.ts]: what's the consequence of this change?

File: ts/sagas/authentication.ts [hunk 2]
```diff
@@ -73,6 +89,71 @@ export function* loadSession(): IterableIterator<Effect> {
...
+      yield put(
+        NavigationActions.navigate({
+          routeName: ROUTES.BACKGROUND
+        })
+      );
+    } else if (
+      lastState === "background" && // The app was in background
+      newState === "active" // The app is now active
+    ) {
+      if (timeElapsed > backgroundActivityTimeout * 1000) {
```
Comment (cloudify) [inline ts/sagas/authentication.ts]: it seems like you can move the computation of `timeElapsed` here since it's used only inside this `else` block

**Pull request merged**